tests/Internal/OktaApiClient.Tests.ps1

```diff
@@ -5,15 +5,16 @@ Context 'Invoke-OktaApiClient - 429 Responses' {
         $Config.RequestTimeout = $null
         $Now = Get-Date # Used as a reference for the test. Indicates when the request was executed
         $ResetDate = $Now.AddSeconds(3) # Indicates when one should retry
-        
+        $ResetDateEpoch = $ResetDate.ToUniversalTime().Subtract((New-Object DateTime(1970, 1, 1, 0, 0, 0, 0))).TotalSeconds
+
         Mock -ModuleName Okta.PowerShell Get-OktaConfiguration { return $Config } -Verifiable
         
         $Response = [PSCustomObject]@{
             Content = "[]"
             Headers =  @{
                 "Date" = @($Now)
                 "Content-Type" = @('application/json; charset=utf-8')
-                "X-Rate-Limit-Reset" = @($ResetDate)
+                "x-rate-limit-reset" = @($ResetDateEpoch)

```

Comment:
Do we also need to fix the casing (or make the appropriate changes to handle headers in a case-insensitive way) of additional headers such as `x-okta-request-id`? Is `x-rate-limit-reset` the only problematic one?

Reply:
Interesting one, I didn't look at that one because it wasn't erroring - now im questioning my case sensitive testing for sure

Reply:
That is a great question and I didnt do that one because it didn't throw errors on me. Im now questioning myself about the case behavior - Ill retest it on the weekend

Reply:
Apologies for the delay, I started in on this and it got weird, but I finally found it

Heres a screenshot of the debugging and some annotations, 

So whats happening is that inside Invoke-OptaApiClient the $headers that come back from the Invoke are taken from the rawResponse.Headers. These are a Dictionary and the reference to the Dictionary keyvalue pairs is case insensitive as you can see in the debug checks using various casing in the box that points to 1

When this Dictionary is passed in to CalculateDelay the Dictionary is cast to a Hashtable which behaves in a case sensitive behavior - as you can see in the debug results in the box pointing to 2

![image](https://github.com/user-attachments/assets/131869f7-3b93-4ecb-9fb2-acd98a47fcbd)

So the short answer is the casing re `x-okta-request-id` works as is because the problem is occuring in the pass of the headers into the `CalculateDelay` method. I really wish I could figure out why this is the case, but the day job is chewing a lot of time

Reply:
Thanks for sharing your findings, and we appreciate the time you took to look into this!
I'm currently working on other priorities, but I'll get back to this as soon as possible.